docs/development/how_to_add_a_route.md

```diff
@@ -0,0 +1,175 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+The goal of this document is to cover all the necessary steps to add an API route to Parsec.
+
+In summary:
+1. Add the schema describing the new API route
+2. Generate binding code and serialization tests
+3. Implement server-side code (In-memory & PostgreSQL) and tests
+4. Implement client-side code and tests
+
+> Note that you will need a working development environment. See (./README.md).
+## Schema generation
+
+Let's add a new `dummy` route to `invited_cmds`.
+
+> API routes are also called "commands" in Parsec. There are three command families: `anonymous_cmds` (authentication not required), `authenticated_cmds` (authentication required) and `invited_cmds` (invitation required).
+
+In [this folder](../../misc/libparsec/crates/protocol/schema/invited_cmds/) add a `dummy.json5` schema describing the API route.
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "dummy",
+            "fields": [
+                {
+                    "name": "dummy_field",
+                    "type": "String"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "dummy_rep_with_fields",
+                "fields": [
+                    {
+                        "name": "dummy_rep_field",
+                        "type": "String"
+                    }
+                ]
+            },
+            {
+                "status": "dummy_rep_without_fields",
+            }
+        ]
+    }
+]
+
+```
+For a full description of the accepted format see [the parsing/generating script](../../misc/gen_protocol_typings.py).
+
+> Note that the files are .json5. This is to allow for comments inside these files. But the parser we are using only supports .json files and we're removing the comments ourselves. So don't use any other specific feature of the json5 format (trailing commas, I'm looking at you 👀).
+
+Then from the root directory, run the following scripts to generate binding code.
+
+```shell
+python make.py r
+```
+
+This will rebuild the bindings (use option `i` to install) and modify the following files:
+    - `server/parsec/_parsec_pyi/protocol/__init__.pyi`
+    - `server/parsec/_parsec_pyi/protocol/xxxxxx_cmds/v4/__init__.pyi`
+    - `server/tests/common/rpc.py`
+
+```shell
+python misc/gen_protocol_typings.py
+```
+
+This will generate the .pyi files and some helpers for the tests. The generated code is in `/target` directory.
```

Comment:
I'd would remove the `The generated code is in `/target` directory` given we never look into this folder when working with python (once `./make.py r` is run, the python module is automatically copied to `server/parsec/_parsec.cpython-312-x86_64-linux-gnu.so`